ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -47,6 +48,16 @@ import { getSeriesColor } from './utils/palette-gen';
 
 export type TimeSeriesChartProps = PanelProps<TimeSeriesChartOptions>;
 
+// Using an "ALL" value to handle the case on first loading the chart where we
+// want to select all, but do not want all of the legend items to be visually highlighted.
+// This helps us differentiate those cases more clearly instead of inferring it
+// based on the state of the data. This also helps us avoid some coding
+// complexity around initializing a full record for the initial load that would
+// currently require significantly more refactoring of this component.
+// TODO: simplify this if we switch the list-based legend UI to use checkboxes,
+// where we *would* want to visually select all items in this case.
```

Comment:
Some additional context that I thought was probably a bit much to put in here, but useful for in flight work:
- The prior implementation inferred "select all" from an empty selected series. This works great for the list legend where there is no "select none," but will become a problem for the table legend where the user can have a "select none" state because it uses checkboxes.
- Initializing a full record of "true" values for the "select all" state is tricky with how all of the data is managed in this component right now. We'd really benefit from breaking it out into 2-3 components, so that we can separate data fetching, transforming the data for rendering, and interaction state initialized from the prior two (e.g. selected items). I really didn't want to take that refactor on while other people are touching this component. Even if I did, it wouldn't address the case of not wanting to select all the list legend items in the initial "select all" state.

Reply:
Love the idea about splitting up into multiple components the way you described